clustered-indexes.md

```diff
@@ -0,0 +1,188 @@
++-------------+---------+------+-------------------------------------------------------+---------------+
+| Point_Get_1 | 1.00    | root | table:will_now_cluster, clustered index:PRIMARY(guid) |               |
++-------------+---------+------+-------------------------------------------------------+---------------+
+1 row in set (0.00 sec)
+
++--------------------------+---------+-----------+------------------------------------+-------------------------------------------------------+
+| id                       | estRows | task      | access object                      | operator info                                         |
++--------------------------+---------+-----------+------------------------------------+-------------------------------------------------------+
+| Projection_4             | 10.00   | root      |                                    | test.will_now_cluster.guid                            |
+| └─IndexReader_6          | 10.00   | root      |                                    | index:IndexRangeScan_5                                |
+|   └─IndexRangeScan_5     | 10.00   | cop[tikv] | table:will_now_cluster, index:b(b) | range:["aaaa","aaaa"], keep order:false, stats:pseudo |
++--------------------------+---------+-----------+------------------------------------+-------------------------------------------------------+
+3 rows in set (0.00 sec)
+```
+
+Clustering by a composite `PRIMARY KEY` is also supported:
+
+```sql
+SET tidb_enable_clustered_index = 1;
+CREATE TABLE composite_primary_key (
+ key_a INT NOT NULL,
+ key_b INT NOT NULL,
+ b CHAR(100),
+ PRIMARY KEY (key_a, key_b)
+);
+
+INSERT INTO composite_primary_key VALUES (1, 1, 'aaa'), (2, 2, 'bbb');
+EXPLAIN SELECT * FROM composite_primary_key WHERE key_a = 1 AND key_b = 2;
+```
+
+```sql
+Query OK, 0 rows affected (0.00 sec)
+
+Query OK, 0 rows affected (0.09 sec)
+
+Query OK, 2 rows affected (0.02 sec)
+Records: 2  Duplicates: 0  Warnings: 0
+
++-------------+---------+------+--------------------------------------------------------------------+---------------+
+| id          | estRows | task | access object                                                      | operator info |
++-------------+---------+------+--------------------------------------------------------------------+---------------+
+| Point_Get_1 | 1.00    | root | table:composite_primary_key, clustered index:PRIMARY(key_a, key_b) |               |
++-------------+---------+------+--------------------------------------------------------------------+---------------+
+1 row in set (0.00 sec)
+```
+
+This behavior is consistent with MySQL, where the InnoDB storage engine will by default cluster by any `PRIMARY KEY`.
+
+## Storage considerations
+
+Because the `PRIMARY KEY` replaces a 64-bit `handle` as the internal pointer to table rows, using clustered indexes might increase storage requirements. This is particularly impactful on tables that contain many secondary indexes. Consider the following example:
```

Comment:
```suggestion
Because the `PRIMARY KEY` replaces a 64-bit `handle` value as the internal pointer to table rows, using clustered indexes might increase storage requirements. This is particularly impactful on tables that contain many secondary indexes. Consider the following example:
```